tests/server.py

```diff
@@ -593,7 +593,7 @@ def flush(self, maxbytes=None):
         if self.disconnected:
             return
 
-        if getattr(self.other, "transport") is None:
+        if self.other.transport is None:
```

Comment:
I'm not sure this change is correct, transport gets injected and might not exist on `self.other`. I'm unsure if that can *actually* happen in practice or not though.

Reply:
`getattr` is the exact same as `.`-access, though, there's absolutely no difference, both (would) still call the internal `__getattr__` method on the object, if it were to exist.

Reply:
I had the same thought, but I think it's an equivalent transformation. Perhaps the original logic is wrong?

```pycon
>>> help(getattr)
Help on built-in function getattr in module builtins:

getattr(...)
    getattr(object, name[, default]) -> value
    
    Get a named attribute from an object; getattr(x, 'y') is equivalent to x.y.
    When a default argument is given, it is returned when the attribute doesn't
    exist; without it, an exception is raised in that case.
```

Reply:
Perhaps the intent was something like:

```python
if not hasattr(self.other, 'transport') or self.other.transport is None:
```

or

```
if not getattr(self.other, 'transport', False):
```

Reply:
getattr is being used to check if an attribute exists, without specifying a default value. Missing a default to getattr will cause an AttributeError to be raised for non-existent properties, which is the same as when a non-existent property is accessed directly. It is recommended to either provide a default value to be returned by getattr if the attribute is not found, or access the attribute directly as there is no additional safety in using getattr if the attribute name is known ahead of time.

Reply:
@clokep , @callahad , @ShadowJonathan thanks for pointing out , I have updated the PR.

Reply:
I think the original intent was `getattr(selfother, 'transport', None)`. But I suppose if we're never hitting this now then maybe it is OK. 🤷